src/freehand_selection.jl

```diff
@@ -1,61 +1,188 @@
 function init_freehand_select(ctx::ImageContext)
     c = ctx.canvas
     enabled = Signal(true)
-    dragging = Signal(false)
+    drawing = Signal(false)
+    moving = Signal(false)
+    diff = Signal(XY(NaN,NaN))
 
     dummybtn = MouseButton{UserUnit}()
     
     sigstart = map(filterwhen(enabled, dummybtn, c.mouse.buttonpress)) do btn
         if !ispolygon(value(ctx.points)) || !isinside(Point(btn.position), Point.(value(ctx.points))) # prevents conflict with init_move_polygon
-            push!(dragging, true)
+            push!(drawing, true)
             push!(ctx.shape, Rectangle()) # some identifier of type of selection
             push!(ctx.points, [])
```

Comment:
> The reason why I wouldn't change it to XY{Float64} is because I also want it to be able to hold XY{UserUnit}. Would XY{Number} still be abstract?

Ah, I see what you're concerned about. If you can guarantee that it will be `UserUnit`, then just say `XY{UserUnit}`. If it could be either type, then you might need to be a little more creative (e.g., `T = typeof(pt); points = T[]` if you have a `pt` that is a "template" for what to expect). Or, if you have to, you can keep it abstract---performance isn't everything, and especially in GUIs flexibility is a good thing.

Somewhat confusingly, `XY{Number}` is concrete but it's element type is not. In practice therefore it will behave like an abstract type.